pkg/informer/informer.go

```diff
@@ -412,35 +420,47 @@ func gvrFor(group, version, resource string) schema.GroupVersionResource {
 	}
 }
 
-func builtInInformableTypes() map[schema.GroupVersionResource]struct{} {
+func withCRDSpec(scope apiextensionsv1.ResourceScope, kind, singular string) apiextensionsv1.CustomResourceDefinitionSpec {
+	return apiextensionsv1.CustomResourceDefinitionSpec{
+		Scope: scope,
+		Names: apiextensionsv1.CustomResourceDefinitionNames{
+			Kind:     kind,
+			Singular: singular,
+		},
+	}
+}
+
+func builtInInformableTypes() map[schema.GroupVersionResource]apiextensionsv1.CustomResourceDefinitionSpec {
 	// Hard-code built in types that support list+watch
-	latest := map[schema.GroupVersionResource]struct{}{
-		gvrFor("", "v1", "configmaps"):                                                  {},
-		gvrFor("", "v1", "events"):                                                      {},
-		gvrFor("", "v1", "limitranges"):                                                 {},
-		gvrFor("", "v1", "namespaces"):                                                  {},
-		gvrFor("", "v1", "resourcequotas"):                                              {},
-		gvrFor("", "v1", "secrets"):                                                     {},
-		gvrFor("", "v1", "serviceaccounts"):                                             {},
-		gvrFor("certificates.k8s.io", "v1", "certificatesigningrequests"):               {},
-		gvrFor("coordination.k8s.io", "v1", "leases"):                                   {},
-		gvrFor("rbac.authorization.k8s.io", "v1", "clusterroles"):                       {},
-		gvrFor("rbac.authorization.k8s.io", "v1", "clusterrolebindings"):                {},
-		gvrFor("rbac.authorization.k8s.io", "v1", "roles"):                              {},
-		gvrFor("rbac.authorization.k8s.io", "v1", "rolebindings"):                       {},
-		gvrFor("events.k8s.io", "v1", "events"):                                         {},
-		gvrFor("admissionregistration.k8s.io", "v1", "mutatingwebhookconfigurations"):   {},
-		gvrFor("admissionregistration.k8s.io", "v1", "validatingwebhookconfigurations"): {},
-		gvrFor("apiextensions.k8s.io", "v1", "customresourcedefinitions"):               {},
-	}
-
-	return latest
+	return map[schema.GroupVersionResource]apiextensionsv1.CustomResourceDefinitionSpec{
```

Comment:
Using a struct (`apiextensionsv1.CustomResourceDefinitionSpec`) that supports many fields, but only ever setting a few of them, seems tricky for follow-up authors to remember & reason about. Why not some super-set type on GVR

```go
type GVRWithExtraInfo {
    schema.GroupVersionResource
    Kind, Singular string
    Scope
}
```

Then you can keep the previous factoring as well

Reply:
Done 👍🏼.